modules/edit-modes/src/lib/edit-mode.js

```diff
@@ -1,118 +1,26 @@
 // @flow
 
 import type {
-  EditAction,
   ClickEvent,
   PointerMoveEvent,
   StartDraggingEvent,
-  StopDraggingEvent
+  StopDraggingEvent,
+  ModeProps
 } from '../types.js';
 
-export type ModeState<TData, TGuides> = {
-  // The data being edited, this can be an array or an object
-  data: TData,
-
-  // Additional configuration for this mode
-  modeConfig: any,
-
-  // The indexes of the selected features
-  selectedIndexes: number[],
-
-  // Features that can be used as a guide for editing the data
-  guides: ?TGuides,
-
-  // The cursor type, as a [CSS Cursor](https://developer.mozilla.org/en-US/docs/Web/CSS/cursor)
-  cursor: string,
-
-  // Callback used to notify applications of an edit action
-  onEdit: (editAction: EditAction<TData>) => void,
-
-  // Callback used to update guides
-  onUpdateGuides: (guides: ?TGuides) => void,
-
-  // Callback used to update cursor
-  onUpdateCursor: (cursor: string) => void
-};
-
 export interface EditMode<TData, TGuides> {
```

Comment:
```_guides: FeatureCollection```, // preserving last guides

Reply:
I think we want to keep `EditMode` more stateless and functional. So, I don't think we want to keep track of the last guides in the state of the mode. Also, `EditMode` is not specific to GeoJSON. For example, we may have an `EditMode` in the future for editing h3 cells.

Memoizing guides to avoid re-calculating them unnecessarily is something that will likely be specialized in each mode. For example, some modes will have guides based on the data alone, other modes need to take the pointer position into account.